### PR TITLE
fix(ast): Fix `StaticMemberExpression.get_first_object`

### PR DESCRIPTION
**Pull request by tbashiyy**
I think `get_first_object` does not return expected expression.

For example, in case of `foo.bar.a`, it doesn't return `foo` but `foo.bar`.

**Expected**
```
{
  "type": "Identifier",
  "start": 0,
  "end": 3,
  "name": "foo"
}
```

**Actual**
```
{
   "type":"StaticMemberExpression",
   "start":0,
   "end":7,
   "object":{
      "type":"Identifier",
      "start":0,
      "end":3,
      "name":"foo"
   },
   "property":{
      "type":"Identifier",
      "start":4,
      "end":7,
      "name":"bar"
   },
   "optional":false
}
```

### CODE REVIEW

**Comment by Boshen** [comment]
@Dunqing can you check the ID failures?

**Review by Dunqing — DISMISSED** [review]

File: crates/oxc_ast/src/ast_impl/js.rs [hunk 1]
```diff
@@ -517,7 +517,7 @@ impl<'a> StaticMemberExpression<'a> {
                 if let Expression::StaticMemberExpression(expr) = &member.object {
                     expr.get_first_object()
                 } else {
-                    &self.object
+                    member.get_first_object()
                 }
             }
```
[the remark this answers is not in the document]
Comment (tbashiyy) [reply crates/oxc_ast/src/ast_impl/js.rs]: Thanks!
fixed 9ae5f1922d687c19deb726d3fa7463eea2358494

**Comment by Boshen** [comment]
Conformance still failing @Dunqing

**Review by Dunqing — APPROVED** [review]
I have checked in [TypeScript](https://www.typescriptlang.org/play/?target=99&isolatedDeclarations=true&noCheck=true&ssl=7&ssc=1&pln=8&pc=1#code/FAUwHgDg9gTgLgAjgTwiBBBBBeBBvYBBAbQFsBLAZ0vIDsBzAXQC4FaBXUgIxBgBpCJWpQB0FanSasO3XgKLEIMEJRC04AOVEBDFm049+g4gGVk3KABsR5OL21xYemYfkl6lqF22WAKgAsqETMLa0cTOBhJX20pfVkjBQAKDy8fAKCQrisAShF2YQBjKAhtLksVZwM5Yx9ybRoGG0oAYShaQocTJRBtABMyiqqEt2IARmHXYwAiACZpyZrk8Ub6WZzFxJIAWQd-ERhtWj6oUiSchAA+BAAGEQBWBAB+BGmAMzHphFZ3+c2BAC+AG5gEA), the snapshot changes is correct. You even fixed an IsolatedDeclarations bug! Thanks!

**Pull request merged**